src/util/animate.ts

```diff
@@ -71,8 +163,9 @@ export function animate(options = {}) {
     delay = 0,
   } = options;
 
-  const context = {
+  const context: Partial<AnimationContext> = {
     ...options,
+    cancel: noop, // placeholder
```

Comment:
can we put the cancel logic here? Instead of assinging it later on

Reply:
yeah I think so